x-pack/legacy/plugins/maps/server/maps_telemetry/maps_telemetry.js

```diff
@@ -32,8 +32,16 @@ function getUniqueLayerCounts(layerCountsList, mapsCount) {
   }, {});
 }
 
-export function buildMapsTelemetry(savedObjects, settings) {
-  const layerLists = savedObjects
+function getIndexPatternsWithGeoFieldCount(indexPatterns) {
+  const fieldLists = indexPatterns.map(indexPattern => JSON.parse(indexPattern.attributes.fields));
```

Comment:
Do you want to add a try/catch block around this? `JSON.parse` could throw and we should handle errors locally here

Reply:
the fields-string is machine-generated from the savedObjectService